src/drivers/drivers.jl

```diff
@@ -85,9 +86,24 @@ function set_driver!(p::Properties, ::POSIX)
 end
 
 function __init__()
-    DRIVERS[API.h5fd_core_init()] = Core
-    DRIVERS[API.h5fd_sec2_init()] = POSIX
-    @require MPI="da04e1cc-30fd-572f-bb4f-1f8673147195" include("mpio.jl")
+    # disable file locking as that can cause problems with mmap'ing
+    if !haskey(ENV, "HDF5_USE_FILE_LOCKING")
+        ENV["HDF5_USE_FILE_LOCKING"] = "FALSE"
+    end
+
+    fapl = HDF5.init!(HDF5.FileAccessProperties())
+    API.h5p_set_fapl_core(fapl, 4096, false)
+    DRIVERS[API.h5p_get_driver(fapl)] = Core
```

Comment:
we could lazily initialize `Core` as well?

Reply:
This way seemed closer to the original, so it's a slightly smaller change, but I don't think there would be a problem lazily initializing Core as well if that's the consensus.

Reply:
Perhaps all the `set_driver!` methods should uniformly  register their hid_t in `DRIVERS`. From this `__init__`, we just call `set_driver!(fapl, POSIX())`.